development/auto-changelog.js

```diff
@@ -67,11 +70,28 @@ async function main() {
   const changelogFilename = path.resolve(__dirname, '..', 'CHANGELOG.md');
   const changelog = await fs.readFile(changelogFilename, { encoding: 'utf8' });
   const changelogLines = changelog.split('\n');
-  const releaseHeaderIndex = changelogLines.findIndex(
-    (line) => line === '## Current Develop Branch',
+
+  // remove the "v" prefix
+  const mostRecentVersion = mostRecentTag.slice(1);
+
+  const isReleaseCandidate = mostRecentVersion !== version;
+  const versionHeader = `## ${version}`;
+  const currentDevelopBranchHeader = '## Current Develop Branch';
+  const currentReleaseHeaderPattern = isReleaseCandidate
+    ? // This ensures this doesn't match on a version with a suffix
+      // e.g. v9.0.0 should not match on the header v9.0.0-beta.0
+      `${versionHeader}$|${versionHeader}\\s`
+    : currentDevelopBranchHeader;
+
+  const releaseHeaderIndex = changelogLines.findIndex((line) =>
+    line.match(new RegExp(currentReleaseHeaderPattern, 'um')),
```

Comment:
Nit: I added the `m` flag locally when testing, trying to handle the `-beta.0` suffix issue, but forgot to remove it. I don't believe it's needed.